src/main/java/nextstep/subway/domain/Sections.java

```diff
@@ -0,0 +1,98 @@
+package nextstep.subway.domain;
+
+import java.util.ArrayList;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+import javax.persistence.CascadeType;
+import javax.persistence.Embeddable;
+import javax.persistence.ForeignKey;
+import javax.persistence.JoinColumn;
+import javax.persistence.OneToMany;
+
+@Embeddable
+public class Sections {
+    @OneToMany(cascade = CascadeType.ALL, orphanRemoval = true)
+    @JoinColumn(name = "line_id", foreignKey = @ForeignKey(name = "fk_line_station_to_line"))
+    private List<Section> sections = new ArrayList<>();
+
+    public void add(final Section section) {
+        validateDuplicate(section);
+        sections.stream()
+                .filter(origin -> origin.intersects(section))
+                .findFirst()
+                .ifPresent(origin -> origin.calculate(section));
+
+        sections.add(section);
+    }
+
+    private void validateDuplicate(final Section section) {
+        if (sections.stream().anyMatch(origin -> origin.equalsStations(section))) {
+            throw new SectionException("이미 등록되어 있는 구간입니다.");
+        }
+    }
+
+    public Set<Station> getStationsOrderBy() {
+        final LinkedHashSet<Station> hashSet = new LinkedHashSet<>();
```

Comment:
다형성을 위해 `Set`을 사용하면 좋을 것 같아요.
```suggestion
        final Set<Station> hashSet = new LinkedHashSet<>();
```

Reply:
순서 유지를 위해 `LinkedHashSet` 구현체를 사용하신 것 같은데.
`List`를 사용하면 순서를 유지할 수 있습니다.
`List`와 `Set` 자료구조의 차이를 알아보면 좋을 것 같아요 ☺️

Reply:
혹시라도 중복되는 데이터가 들어가지 못하도록 `Set` 을 사용했는데 순서 유지 를 위해 `LinkedHashSet`을 사용해서 처리했습니다.
사실 구간 추가하는 부분에서 이미 중복되지 못하도록 구성을 해서 `List` 를 사용해도 문제는 없을 거 같긴합니다. 👍